src/components/Form/Form.tsx

```diff
@@ -3,47 +3,89 @@ import {
 	StyledDiv,
 	StyledFieldset,
 	StyledInput,
+	StyledLabel,
 	StyledLegend,
 } from "./Form.Styled.ts";
 import { useState } from "react";
 import NavLinkButton from "../Button/NavLinkButton.tsx";
+import InputError from "./InputError.tsx";
+import { useNavigate } from "react-router-dom";
 
 function LoginForm() {
 	const [password, setPassword] = useState("");
-	const [name, setName] = useState("");
+	const [email, setEmail] = useState("");
+	const [error, setError] = useState<{ email?: string; password?: string }>({});
+	const navigate = useNavigate();
+
+	const validateEmail = (email: string) => {
+		const validEmail = new RegExp(
+			"^[a-zA-Z0-9._:$!%-]+@[a-zA-Z0-9.-]+.[a-zA-Z]$"
+		);
+		return validEmail.test(email);
+	};
+
+	const validatePassword = (password: string) => {
+		return password.length >= 6;
+	};
+
+	const handleSubmit = (e: React.FormEvent) => {
+		console.log(e.target);
+		e.preventDefault();
+		const newError: { email?: string; password?: string } = {};
+
+		if (!validateEmail(email)) {
+			newError.email = "Email format is wrong";
+		}
+		if (!validatePassword(password)) {
+			newError.password = "Password is to short. Min. 6 signs";
+		}
+
+		setError(newError);
+		console.log(typeof newError);
+
+		navigate("/filmlist");
+
+		// if (Object.keys(newError).length === 0) {
+		// 	navigate("/filmlist");
+		// }
+	};
 
 	return (
-		<Form
-			onSubmit={(e) => {
-				e.preventDefault();
-			}}
-		>
+		<Form id="form" onSubmit={handleSubmit}>
 			<StyledFieldset>
 				<StyledLegend>Zaloguj się</StyledLegend>
 				<StyledDiv>
+					<StyledLabel htmlFor="email">Email*</StyledLabel>
 					<StyledInput
+						id="email"
 						type="text"
 						placeholder="Email"
-						value={name}
-						onChange={(e) => setName(e.target.value)}
+						value={email}
+						onChange={(e) => setEmail(e.target.value)}
 						required
 					/>
+					<InputError text={error.email || ""} error={!!error.email} />
```

Comment:
Otypuj text jako ? nullable, wtedy tutaj przekażesz to bez || ""

A najlepiej jak zrobisz jeden atrybut: error={error.email} i on w srodku sobie sprawdzi czy jest tresc (jako boolean go sprawdzisz) i będziesz miał tylko jeden props